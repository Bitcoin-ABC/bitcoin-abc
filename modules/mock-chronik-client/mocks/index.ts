// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import {
    Block,
    BlockchainInfo,
    Tx,
    TokenInfo,
    ScriptUtxo,
    Utxo,
} from 'chronik-client';

interface ChronikMocks {
    tipHeight: number;
    tipHash: string;
    block: Block;
    blockchainInfo: BlockchainInfo;
    txid: string;
    tx: Tx;
    tokenId: string;
    token: TokenInfo;
    rawTx: string;
    scriptUtxo: ScriptUtxo;
    utxo: Utxo;
}
const DUMMY_SCRIPT_UTXO: ScriptUtxo = {
    outpoint: { txid: '00'.repeat(32), outIdx: 0 },
    blockHeight: 800000,
    isCoinbase: false,
    value: 546,
    isFinal: true,
};
const DUMMY_UTXO: Utxo = {
    outpoint: { txid: '00'.repeat(32), outIdx: 0 },
    script: `76a914${'00'.repeat(20)}88ac`,
    blockHeight: 800000,
    isCoinbase: false,
    value: 546,
    isFinal: true,
};
const chronikMocks: ChronikMocks = {
    tipHeight: 800000,
    tipHash: '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
    block: {
        blockInfo: {
            hash: '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
            prevHash:
                '0000000000000000023ab587190a05e44cdb76fe1dd7949f6187f1e632d2e123',
            height: 800000,
            nBits: 403943060,
            timestamp: 1688808780,
            isFinal: true,
            blockSize: 3094,
            numTxs: 9,
            numInputs: 13,
            numOutputs: 26,
            sumInputSats: 859760862,
            sumCoinbaseOutputSats: 625005728,
            sumNormalOutputSats: 859755134,
            sumBurnedSats: 0,
        },
    },
    blockchainInfo: {
        tipHash:
            '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
        tipHeight: 800000,
    },
    txid: '2c1540131fb7611b72ec3ca9a215ac5c408248327d3c5b5ff987cd2a959ee1d2',
    tx: {
        txid: '2c1540131fb7611b72ec3ca9a215ac5c408248327d3c5b5ff987cd2a959ee1d2',
        version: 2,
        inputs: [
            {
                prevOut: {
                    txid: '1cb37ece9d9dbc19ef2c6d57766ca3b91bdc3e56d203aa0fc1b41a7268c0c9d2',
                    outIdx: 2,
                },
                inputScript:
                    '41ebe634094888c4215b8690546b74695a048a9c49c8a5c6efedd940591ec9039e91cf2a6f7836107afbc41bb6e13fb22d99739aeadf1a740f75efea177cfcd86641210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7c',
                value: 546,
                sequenceNo: 4294967295,
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '455320000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
            },
            {
                prevOut: {
                    txid: '1cb37ece9d9dbc19ef2c6d57766ca3b91bdc3e56d203aa0fc1b41a7268c0c9d2',
                    outIdx: 3,
                },
                inputScript:
                    '412e7de41834aaf15d16347a59c954cb886d8ec00584bb566252e16f8e1779ce2e4d6ddd91782bd6ba2f695a28ce4115cd0561d27c1ae16b3d8bf77f8186235b1441210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7c',
                value: 94518653,
                sequenceNo: 4294967295,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
            },
        ],
        outputs: [
            {
                value: 0,
                outputScript:
                    '6a04534c500001010453454e4420aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb108000000000000271008000000001b237ab0',
            },
            {
                value: 546,
                outputScript:
                    '76a914a805a320360fa685f83605d8e56de6f9d8a7a99988ac',
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '10000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
            },
            {
                value: 546,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '455310000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
                spentBy: {
                    txid: 'b334ad4ee98718707142a7a951163f1cb94817269b55ac840a7b416ce256eecd',
                    outIdx: 0,
                },
            },
            {
                value: 94517640,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
                spentBy: {
                    txid: 'b334ad4ee98718707142a7a951163f1cb94817269b55ac840a7b416ce256eecd',
                    outIdx: 1,
                },
            },
        ],
        lockTime: 0,
        timeFirstSeen: 0,
        size: 467,
        isCoinbase: false,
        tokenEntries: [
            {
                tokenId:
                    'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                tokenType: {
                    protocol: 'SLP',
                    type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                    number: 1,
                },
                txType: 'SEND',
                isInvalid: false,
                burnSummary: '',
                failedColorings: [],
                actualBurnAmount: '0',
                intentionalBurn: '0',
                burnsMintBatons: false,
            },
        ],
        tokenFailedParsings: [],
        tokenStatus: 'TOKEN_STATUS_NORMAL',
        isFinal: false,
        block: {
            height: 866581,
            hash: '000000000000000000860f542fac8cbb10614e40ab0f37f90ec4d41b126eb5d9',
            timestamp: 1728930658,
        },
    },
    tokenId: 'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
    token: {
        tokenId:
            'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
        tokenType: {
            protocol: 'SLP',
            type: 'SLP_TOKEN_TYPE_FUNGIBLE',
            number: 1,
        },
        timeFirstSeen: 1711776546,
        genesisInfo: {
            tokenTicker: 'CACHET',
            tokenName: 'Cachet',
            url: 'https://cashtab.com/',
            decimals: 2,
            hash: '',
        },
        block: {
            height: 838192,
            hash: '0000000000000000132232769161d6211f7e6e20cf63b26e5148890aacd26962',
            timestamp: 1711779364,
        },
    },
    rawTx: '0200000002d2c9c068721ab4c10faa03d2563edc1bb9a36c76576d2cef19bc9d9dce7eb31c020000006441ebe634094888c4215b8690546b74695a048a9c49c8a5c6efedd940591ec9039e91cf2a6f7836107afbc41bb6e13fb22d99739aeadf1a740f75efea177cfcd86641210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7cffffffffd2c9c068721ab4c10faa03d2563edc1bb9a36c76576d2cef19bc9d9dce7eb31c0300000064412e7de41834aaf15d16347a59c954cb886d8ec00584bb566252e16f8e1779ce2e4d6ddd91782bd6ba2f695a28ce4115cd0561d27c1ae16b3d8bf77f8186235b1441210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7cffffffff040000000000000000406a04534c500001010453454e4420aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb108000000000000271008000000001b237ab022020000000000001976a914a805a320360fa685f83605d8e56de6f9d8a7a99988ac22020000000000001976a914821407ac2993f8684227004f4086082f3f801da788ac8839a205000000001976a914821407ac2993f8684227004f4086082f3f801da788ac00000000',
    scriptUtxo: DUMMY_SCRIPT_UTXO,
    utxo: DUMMY_UTXO,
};

export default chronikMocks;
