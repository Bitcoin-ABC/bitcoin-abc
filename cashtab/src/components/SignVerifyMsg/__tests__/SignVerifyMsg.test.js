// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { walletWithXecAndTokens } from 'components/App/fixtures/mocks';
import { when } from 'jest-when';
import 'fake-indexeddb/auto';
import localforage from 'localforage';
import appConfig from 'config/app';
import {
    initializeCashtabStateForTests,
    clearLocalForage,
} from 'components/App/fixtures/helpers';
import CashtabTestWrapper from 'components/App/fixtures/CashtabTestWrapper';

// https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function
Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

// https://stackoverflow.com/questions/64813447/cannot-read-property-addlistener-of-undefined-react-testing-library
window.matchMedia = query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
});

describe('<SignVerifyMsg />', () => {
    let user;
    beforeEach(() => {
        // Set up userEvent
        user = userEvent.setup();
        // Mock the fetch call for Cashtab's price API
        global.fetch = jest.fn();
        const fiatCode = 'usd'; // Use usd until you mock getting settings from localforage
        const cryptoId = appConfig.coingeckoId;
        // Keep this in the code, because different URLs will have different outputs requiring different parsing
        const priceApiUrl = `https://api.coingecko.com/api/v3/simple/price?ids=${cryptoId}&vs_currencies=${fiatCode}&include_last_updated_at=true`;
        const xecPrice = 0.00003;
        const priceResponse = {
            ecash: {
                usd: xecPrice,
                last_updated_at: 1706644626,
            },
        };
        when(fetch)
            .calledWith(priceApiUrl)
            .mockResolvedValue({
                json: () => Promise.resolve(priceResponse),
            });
    });
    afterEach(async () => {
        jest.clearAllMocks();
        await clearLocalForage(localforage);
    });
    it('Notification is rendered upon successfully signing a message', async () => {
        // Mock the app with context at the SignVerifyMsg screen
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithXecAndTokens,
            localforage,
        );
        render(
            <CashtabTestWrapper
                chronik={mockedChronik}
                route="/signverifymsg"
            />,
        );

        // Wait for the app to load
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // Insert message to be signed
        await user.type(
            screen.getByPlaceholderText('Enter message to sign'),
            'test message',
        );

        // Click the Sign button (the Switch is also found by this identifier, btn is at index 0 here)
        await user.click(screen.getAllByRole('button', { name: /Sign/ })[0]);

        expect(await screen.findByText('Message Signed')).toBeInTheDocument();

        expect(
            screen.getByText(
                'ILwXI6gAnIkh7nw3r/VDNYWiBxVoHYIjsk9lALezbYjGVXbFj0p4glXibHDQoJVll4+zTS5SWXsVAwhczh5GTts=',
            ),
        ).toBeInTheDocument();
    });
    it('Notification is rendered upon successfully verifying a message', async () => {
        // Mock the app with context at the SignVerifyMsg screen
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithXecAndTokens,
            localforage,
        );
        render(
            <CashtabTestWrapper
                chronik={mockedChronik}
                route="/signverifymsg"
            />,
        );

        // Wait for the app to load
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // Click the switch to show verify forms
        await user.click(screen.getByTitle('Toggle Sign Verify'));

        // Insert message to be signed
        await user.type(
            await screen.findByPlaceholderText('Enter message to verify'),
            'test message',
        );

        // Input the address
        await user.type(
            screen.getByPlaceholderText('Enter address of signature to verify'),
            'ecash:qq3spmxfh9ct0v3vkxncwk4sr2ld9vkhgvlu32e43c',
        );

        // Insert signature in Signature textarea of Verify collapse
        await user.type(
            screen.getByPlaceholderText('Enter signature to verify'),
            'H6Rde63iJ93n/I7gUac/xheY3mL1eAt2uIR54fgre6O3Om8ogWe+DASNQGDDBkNY43JIGwAIPq9lmMJjeykYFNQ=',
        );

        // Click the Verify button
        // react testing library also finds the switch with this, button is at index 0
        await user.click(screen.getAllByRole('button', { name: /Verify/ })[0]);

        expect(
            screen.getByText(
                'Signature verified. Message "test message" was signed by ecash:qq3spmxfh9ct0v3vkxncwk4sr2ld9vkhgvlu32e43c',
            ),
        ).toBeInTheDocument();
    });
    it('Notification is rendered upon signature verification error', async () => {
        // Mock the app with context at the SignVerifyMsg screen
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithXecAndTokens,
            localforage,
        );
        render(
            <CashtabTestWrapper
                chronik={mockedChronik}
                route="/signverifymsg"
            />,
        );

        // Wait for the app to load
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // Click the switch to show verify forms
        await user.click(screen.getByTitle('Toggle Sign Verify'));

        // Insert message to be signed
        await user.type(
            screen.getByPlaceholderText('Enter message to verify'),
            'NOT THE RIGHT MESSAGE',
        );

        // Input the address
        await user.type(
            screen.getByPlaceholderText('Enter address of signature to verify'),
            'ecash:qq3spmxfh9ct0v3vkxncwk4sr2ld9vkhgvlu32e43c',
        );

        // Insert signature in Signature textarea of Verify collapse
        await user.type(
            screen.getByPlaceholderText('Enter signature to verify'),
            'H6Rde63iJ93n/I7gUac/xheY3mL1eAt2uIR54fgre6O3Om8ogWe+DASNQGDDBkNY43JIGwAIPq9lmMJjeykYFNQ=',
        );

        // Click the Verify button
        // react testing library also finds the switch with this, button is at index 0
        await user.click(screen.getAllByRole('button', { name: /Verify/ })[0]);

        expect(
            screen.getByText('Signature does not match address and message'),
        ).toBeInTheDocument();
    });
});
