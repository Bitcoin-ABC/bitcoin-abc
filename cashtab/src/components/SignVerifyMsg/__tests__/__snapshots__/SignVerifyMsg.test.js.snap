// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet with BCH balances 1`] = `
<div
  className="sc-hmzhuo dORcxa"
>
  <div
    className="sc-iAyFgw gXtUud"
  >
    <div
      className="sc-iRbamj ivJrZB"
    >
      <h4
        className="sc-jlyJG isCono"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-cvbbAY ekZlPO"
    >
      You currently have 0 
      XEC
      <br />
      Deposit some funds to use this feature
    </div>
  </div>
   
  <div
    className="sc-kEYyzF gZfJyA"
  >
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Sign
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Verify
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wallet with BCH balances and tokens and state field 1`] = `
<div
  className="sc-hmzhuo dORcxa"
>
  <div
    className="sc-iAyFgw gXtUud"
  >
    <div
      className="sc-iRbamj ivJrZB"
    >
      <h4
        className="sc-jlyJG isCono"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-eHgmQL znFwg"
    >
      <span
        className="sc-gipzik kLBama"
      >
         
        0.06
         
        XEC
         
      </span>
    </div>
  </div>
   
  <div
    className="sc-kEYyzF gZfJyA"
  >
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Sign
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Verify
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wallet without BCH balance 1`] = `
<div
  className="sc-hmzhuo dORcxa"
>
  <div
    className="sc-iAyFgw gXtUud"
  >
    <div
      className="sc-iRbamj ivJrZB"
    >
      <h4
        className="sc-jlyJG isCono"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-cvbbAY ekZlPO"
    >
      You currently have 0 
      XEC
      <br />
      Deposit some funds to use this feature
    </div>
  </div>
   
  <div
    className="sc-kEYyzF gZfJyA"
  >
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Sign
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Verify
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Without wallet defined 1`] = `
<div
  className="sc-hmzhuo dORcxa"
>
  <div
    className="sc-iAyFgw gXtUud"
  >
    <div
      className="sc-iRbamj ivJrZB"
    >
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-cvbbAY ekZlPO"
    >
      You currently have 0 
      XEC
      <br />
      Deposit some funds to use this feature
    </div>
  </div>
   
  <div
    className="sc-kEYyzF gZfJyA"
  >
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Sign
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-jhAzac jaUNeB"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-fMiknA jcDsbP"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-hzDkRC kpsoyz"
              >
                Verify
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
